permissions/resource_access_control_rule_set.go

```diff
@@ -0,0 +1,139 @@
+package permissions
+
+import (
+	"context"
+	"errors"
+	"net/http"
+
+	"github.com/databricks/databricks-sdk-go/apierr"
+	"github.com/databricks/databricks-sdk-go/service/iam"
+	"github.com/databricks/terraform-provider-databricks/common"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceAccessControlRuleSet() *schema.Resource {
+	s := common.StructToSchema(
+		iam.RuleSetUpdateRequest{},
+		func(m map[string]*schema.Schema) map[string]*schema.Schema {
+			m["etag"].Required = false
+			m["etag"].Computed = true
+			m["grant_rules"].Type = schema.TypeSet
+			common.MustSchemaPath(m, "grant_rules", "principals").Type = schema.TypeSet
+
+			return m
+		})
+	readFromWsOrAcc := func(ctx context.Context, c *common.DatabricksClient, getRuleSetReq iam.GetRuleSetRequest) (*iam.RuleSetResponse, error) {
+		if c.Config.AccountID != "" {
+			accountClient, err := c.AccountClient()
+			if err != nil {
+				return nil, err
+			}
+			return accountClient.AccessControl.GetRuleSet(ctx, getRuleSetReq)
+		}
+		workspaceClient, err := c.WorkspaceClient()
+		if err != nil {
+			return nil, err
+		}
+		return workspaceClient.AccountAccessControlProxy.GetRuleSet(ctx, getRuleSetReq)
+	}
+	updateThroughWsOrAcc := func(ctx context.Context, c *common.DatabricksClient, updateRuleSetReq iam.UpdateRuleSetRequest) (*iam.RuleSetResponse, error) {
+		if c.Config.AccountID != "" {
+			accountClient, err := c.AccountClient()
+			if err != nil {
+				return nil, err
+			}
+			return accountClient.AccessControl.UpdateRuleSet(ctx, updateRuleSetReq)
+		}
+		workspaceClient, err := c.WorkspaceClient()
+		if err != nil {
+			return nil, err
+		}
+		return workspaceClient.AccountAccessControlProxy.UpdateRuleSet(ctx, updateRuleSetReq)
+	}
+	fetchLatestEtagAndUpdateRuleSet := func(ctx context.Context, c *common.DatabricksClient, ruleSetUpdateReq iam.UpdateRuleSetRequest) (string, error) {
+		ruleSetGetRes, err := readFromWsOrAcc(ctx, c, iam.GetRuleSetRequest{
+			Name: ruleSetUpdateReq.Name,
+			Etag: "",
+		})
+		if err != nil {
+			return "", err
+		}
+		ruleSetUpdateReq.RuleSet.Etag = ruleSetGetRes.Etag
+		ruleSetUpdateRes, err := updateThroughWsOrAcc(ctx, c, ruleSetUpdateReq)
+		if err != nil {
+			return "", err
+		}
+		return ruleSetUpdateRes.Etag, nil
+	}
+	handleConflictAndUpdate := func(ctx context.Context, c *common.DatabricksClient, ruleSetUpdateReq iam.UpdateRuleSetRequest) (string, error) {
+		ruleSetUpdateRes, err := updateThroughWsOrAcc(ctx, c, ruleSetUpdateReq)
+		if err != nil {
+			var aerr *apierr.APIError
+			if !errors.As(err, &aerr) {
+				return "", err
+			}
+			if aerr.StatusCode == http.StatusConflict {
+				if aerr.ErrorCode == "RESOURCE_CONFLICT" {
+					// we need to get and update
+					etag, err := fetchLatestEtagAndUpdateRuleSet(ctx, c, ruleSetUpdateReq)
```

Comment:
IIUC, this only retries once on resource conflict. Do you want to retry more times ever, or is one retry sufficient?

Reply:
One retry is usually sufficient. Change to the rule set is expected to be an infrequent operation.